server/src/main/java/org/elasticsearch/common/blobstore/BlobPath.java

```diff
@@ -12,38 +12,26 @@
 import org.elasticsearch.common.util.CollectionUtils;
 
 import java.util.Collections;
-import java.util.Iterator;
 import java.util.List;
 import java.util.Objects;
 
 /**
  * The list of paths where a blob can reside.  The contents of the paths are dependent upon the implementation of {@link BlobContainer}.
  */
-public class BlobPath implements Iterable<String> {
```

Comment:
I'm +1 on this change for sure, but I don't see why it's combined with the rest of this PR. Am I missing something, or could you do this separately?

Reply:
:facepalm: yea I I'll back this out here sorry for the noise. I did some magic with `BlobPath` earlier that needed this but ultimately didn't need it after all :)

Reply:
I opened https://github.com/elastic/elasticsearch/pull/72860 :)